exchange_calendars/exchange_calendar.py

```diff
@@ -75,23 +74,87 @@ def _group_times(all_days, times, tz, offset=0):
+    For exchanges that do not observe an intraday break a session
+    represents a contiguous set of minutes. Where an exchange observes
+    an intraday break a session represents two contiguous set of minutes
+    separated by the intraday break.
 
-    A session represents a contiguous set of minutes, and has a label that is
-    midnight UTC. It is important to note that a session label should not be
-    considered a specific point in time, and that midnight UTC is just being
-    used for convenience.
+    Each session has a label that is midnight UTC. It is important to note
+    that a session label should not be considered a specific point in time,
+    and that midnight UTC is just being used for convenience.
 
-    For each session, we store the open and close time in UTC time.
+    For each session, we store the open and close time together with, for
+    those exchanges with breaks, the break start and break end. All times
+    are defined as UTC.
+
+    Parameters
+    ----------
+    start : default: later of 20 years ago or first supported start date.
+        First calendar session will be `start`, if `start` is a session, or
+        first session after `start`.
+
+    end : default: earliest of 1 year from 'today' or last supported end date.
+        Last calendar session will be `end`, if `end` is a session, or last
+        session before `end`.
+
+    Raises
+    ------
+    ValueError
+        If `start` is earlier than the earliest supported start date.
+        If `end` is later than the latest supported end date.
+        If `start` parses to a later date than `end`.
+
+    Notes
+    -----
+    Exchange calendars were originally defined for the Zipline package from
+    Quantopian under the package 'trading_calendars'. Since 2021 they have
+    been maintained under the 'exchange_calendars' package (a fork of
+    'trading_calendars') by an active community of contributing users.
+
+    Some calendars have defined start and end bounds within which
+    contributors have endeavoured to ensure the calendar's accuracy and
+    outside of which the calendar would not be accurate. These bounds
+    are enforced such that passing `start` or `end` as dates that are
+    out-of-bounds will raise a ValueError. The bounds of each calendar are
+    exposed via the `bound_start` and `bound_end` properties.
+
+    Many calendars do not have bounds defined (in these cases `bound_start`
+    and/or `bound_end` return None). These calendars can be created through
+    any date range although it should be noted that the further back the
+    start date, the greater the potential for inaccuracies.
+
+    In all cases, no guarantees are offered as to the accuracy of any
+    calendar.
     """
 
-    def __init__(self, start=start_default, end=end_default):
-        # Midnight in UTC for each trading day.
+    def __init__(self, start: Session | None = None, end: Session | None = None):
```

Comment:
```suggestion
    def __init__(self, start: Optional[Session] = None, end: Optional[Session]  = None):
```

Reply:
The | operator is being introduced for this purpose from Python 3.10 ([PEP 604](https://www.python.org/dev/peps/pep-0604/)). It's available from 3.7 with `from __future__ import annotations` and pylint's already picking it up... `Consider using alternative Union syntax instead of 'Optional'`.

Reply:
I'm going to merge with the | operator, if there's a particular reason for wanting Optional let me know and I'll change it back in a later commit (I believe some static tools aren't up to speed with the new syntax yet).

Reply:
Ah sure - no big deal. Happy with either one.